doc/user/core/image_preprocessing.md

```diff
@@ -0,0 +1,68 @@
+# Image Preprocessing
+
+When loading images using `data::Load()` channels are interleaved, i.e.
+`[r, g, b, r, g, b, ... ]`. mlpack has functionality such as `Convolution`
+that requires channels be grouped, e.g `[r, r, ..., g, g, ..., b, b]`. The
+same is true when using `data::Save()`, the channels are expected to be
+interleaved.
+
+To convert the layout of your image from interleaved channels to grouped
+channels and vice versa, you can use `data::GroupChannels()` and
+`data::InterleaveChannels()`.
+
+---
+
+## `data::GroupChannels()`
+
+ * `data::GroupChannels(images, info)`
+    - `images` must be a matrix where each column is an image. Each image is
+      expected to be interleaved, i.e. in the format `[r, g, b, r, g, b ... ]`.
+
+    - `info` describes the shape of each image.
+
+    - Returns a matrix where each image from `images` are in the
+      format `[r, r, ... , g, g, ... , b, b]`.
```

Comment:
```suggestion
      format `[r, r, ... , g, g, ... , b, b]`.  The size of the matrix is the same as the size of `images`.
```

Technically it's redundant, but I think it's nice to add little things like this so that when a reader reads it, it confirms their understanding.